### ISSUE TITLE
Project is creating in broken state

### ISSUE THREAD

**Issue by saggammahesh** [Adobe-Marketing-Cloud/aem-eclipse-developer-tools#78]
Hi Team,
I  am trying to create a multi module project, but it's failing. 
I am getting an error  "Could not update project training.core.configuration Java.lang.IllegalArgumentException"
![core_issue](https://cloud.githubusercontent.com/assets/7000621/25748647/6a04d88e-31c9-11e7-9d44-2d4ec6bc909e.png)

Below are the different versions i am using. 
Java - 1.8
Eclipse Version: Neon
Maven - Inbuilt one comes with eclipse - 3.3.9

Please let me know what is the issue and how to resolve.

**Comment by rombert** [comment]
Do you have more details in the eclipse error log?

On May 5, 2017 17:01, "saggammahesh" <notifications@github.com> wrote:

Hi Team,
I am trying to create a multi module project, but it's failing.
I am getting an error "Could not update project training.core.configuration
Java.lang.IllegalArgumentException"
[image: core_issue]
<https://cloud.githubusercontent.com/assets/7000621/25748647/6a04d88e-31c9-11e7-9d44-2d4ec6bc909e.png>

Below are the different versions i am using.
Java - 1.8
Eclipse Version: Neon
Maven - Inbuilt one comes with eclipse - 3.3.9

Please let me know what is the issue and how to resolve.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Adobe-Marketing-Cloud/aem-eclipse-developer-tools/issues/78>,
or mute the thread

**Comment by sduvan** [comment]
@rombert  @saggammahesh 
Same issue here, was playing with the new Archetype,  v11.  I am also on Eclipse Neon 3, verified my settings.xml (as per Adobe's recommendation). Issue is also reproducible when project is created through CMD:

> mvn org.apache.maven.plugins:maven-archetype-plugin:2.4:generate -DarchetypeGroupId=com.adobe.granite.archetypes -DarchetypeArtifactId=aem-project-archetype -DarchetypeVersion=11 -DarchetypeCatalog=https://repo.adobe.com/nexus/content/groups/public/

However I can run successful Maven commands outside my IDE (Eclipse).
> **mvn clean install -PautoInstallPackage**

and even the bundle install works through CMD:
> **mvn clean install -PautoInstallBundle**

![aem bundle project sructure](https://cloud.githubusercontent.com/assets/21141902/26126174/7f0f12ee-3a39-11e7-9312-d431be6dbfd1.PNG)

The issue seems to be a IDE related. Attached is the screenshot of my bundle (core) project. It does not reflect a correct Maven structure.

- PS: Verified that Archetype v10 just works fine.

**Comment by akfaircg** [comment]
@rombert,
I faced the same problem. I believe the problem is in the following code snippet that I took from the generated pom.xml in core module in the deceleration of the felix plugin:

`<executions>
      <execution>
           <id>bundle-manifest</id>
           <phase>process-classes</phase>
            <goals>
                 <goal>manifest</goal>
            </goals>
       </execution>
</executions>`

I found the followoing [issue ](https://github.com/tesla/m2eclipse-tycho/issues/4) that I think is related to this one. When I removed the above code and updated the project through maven in ecipse, everything worked fine and I was able to upload the modules to an AEM instance through eclipse.

**Comment by saggammahesh** [comment]
@akfaircg 
Thanks Mate, It started working after removing those few lines.

**Comment by sduvan** [comment]
Also worked for me, thanks!

**Comment by rombert** [comment]
Thanks for tracking down the issue @akfaircg . Would you be willing to submit a pull request for https://github.com/Adobe-Marketing-Cloud/aem-project-archetype/ with a fix?

**Comment by akfaircg** [comment]
@rombert, I just submitted a pull request to https://github.com/Adobe-Marketing-Cloud/aem-project-archetype/. Thanks!

**Comment by rombert** [comment]
Applied, thanks @akfaircg !

**Comment by appsparkler** [comment]
thanks a lot @akfaircg!

@rombert, request you to help me understand.  In this thread you've mentioned _Applied_; however, when i run `mvn archetype:generate.....`, those lines are still there in the `core/pom.xml` file.  Are there any specific steps to get the project without the code-block that is causing the issue?

Thanks & Regards,
Akash

**Comment by rombert** [comment]
@appsparkler - this is not yet included in a release, will be there in version 12 of the archetype.

**Comment by kwin** [comment]
I don't agree with the fix here. Removing the manifest execution has negative sideeffects as unit tests relying on Sling Mocks or AEM Mocks might no longer execute successfully. For details look at http://felix.apache.org/documentation/faqs/apache-felix-bundle-plugin-faq.html#use-scr-metadata-generated-by-bnd-in-unit-tests. 

So the real solution is to get rid of the m2-tycho plugin at all (also compare with https://github.com/tesla/m2eclipse-tycho/issues/31). {{maven-bundle-plugin}} since version 3.2.0 added support for Eclipse natively. This will be fixed in the upcoming version of Sling IDE (https://issues.apache.org/jira/browse/SLING-6112). But in no case the {{manifest}} goal should be completely removed. 

A full explanation of the issue described here can be found in http://sling.apache.org/documentation/development/ide-tooling/ide-tooling-incremental-build.html.

@rombert Can you revert the fix please?
